### PR TITLE
Add proper UTF8 Support

### PR DESCRIPTION
**Pull request by GPSnoopy**
Fixes Issue #205.

Doing this PR requires `[MarshalAs(UnmanagedType.LPUTF8Str)]` which is only supported in net47 or higher. This means I've taken the decision to retire support for .NET Framework 4.6.1 and .NET Core 2.0 or lower. Otherwise, the code would be significantly more complicated and require explicit conversion of all strings to UTF8 before calling into the native methods. Even so, I had to backport `PtrToStringUTF8` from .NET Standard 2.1 (can't believe marshalling UTF8 wasn't properly supported until .NET Core 3.0).

The rest should be less controversial.

One area I'm not sure is `key_id` vs `key_metadata`. I think one of them is meant to be a byte array while the other should be a UTF8 string. But the documentation is lacking. This PR does not address this ambiguity (left out of scope on purpose).

### CODE REVIEW

File: csharp/StringUtil.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace ParquetSharp
+{
+    internal static class StringUtil
+    {
+        public static unsafe IntPtr ToCStringUtf8(string str, ByteBuffer byteBuffer)
+        {
+            var ascii = System.Text.Encoding.UTF8;
```
Comment (adamreeve) [inline csharp/StringUtil.cs:10]: ascii?

Comment (GPSnoopy) [reply csharp/StringUtil.cs:10]: Doh! 🤦

Comment (GPSnoopy) [reply csharp/StringUtil.cs]: Fixed

**Review by adamreeve — APPROVED** [review]

**Review by jgiannuzzi — DISMISSED** [review]
Thanks for the review @adamreeve! 🚀
And thanks for the PR @GPSnoopy!

**Review by jgiannuzzi — APPROVED** [review]

**Pull request merged**